### DISCUSSION TITLE
async idGenerator

### DISCUSSION THREAD

**Discussion by ryanleecode** [oliver-oloughlin/kvdex#195] [Ideas]
Is there any drawbacks to making the `idGenerator` async? For example I might want to use the web crypto api and do a sha256 hash to generate the id but that function is async.

`crypto.subtle.digest('SHA-256', ...)`

**Comment by oliver-oloughlin** [comment]
The reason an id generator cannot be async right now is because it is used when setting new entries within an atomic operation, where the actual set/add calls are non-async. I do however have a theoretical solution to this that I am planning to look into. The exact same restriction applies to serialize/compress functions, which is a blocker for some serialization and compression methods I would like to support.

**Comment by oliver-oloughlin** [comment]
Feel free to open an issue on this btw
